plugins/common/parsing/parsing.go

```diff
@@ -0,0 +1,203 @@
+package parsing
+
+import (
+	"errors"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"strconv"
+	"strings"
+)
+
+var (
+	ErrEmptyDelimiter = errors.New("parsing: delimiter must not be empty")
+	ErrEmptyWildcard  = errors.New("parsing: wildcard must not be empty")
+)
+
+type Config struct {
+	config    []ConfigEntry
+	baseName  string
+	delimiter string
+	wildcard  string
+	log       telegraf.Logger
+}
+
+// NewConfig creates a new parser to extract measurements, tags and fields from structures like
+// mqtt topic or nats subjects.
+// each entry in the config describes how this structure should be interpreted.
+// baseName is just the individual name of the structure (e.g: 'topic' for mqtt).
+// delimiter is the character or string that divides the topic in different parts.
+// wildcard is the character or string that tells that a part of the topic can be any string.
+func NewConfig(entries []ConfigEntry,
+	baseName string,
+	delimiter string,
+	wildcard string,
+	log telegraf.Logger) *Config {
+	return &Config{
+		config:    entries,
+		baseName:  baseName,
+		delimiter: delimiter,
+		wildcard:  wildcard,
+		log:       log,
+	}
+}
```

Comment:
Do we really need this function? I would rather export the `Config` fields and set them directly.